### PR TITLE
[13.0][ENH] account_invoice_margin: add 'Margin(%)' field on Invoice analysis report

### PR DESCRIPTION
**Pull request by bodedra**
Fp: https://github.com/OCA/margin-analysis/pull/102

### CODE REVIEW

**Comment by legalsylvain** [comment]
something is wrong with travis.

**Comment by bodedra** [comment]
> something is wrong with travis.

just fixed it.

**Comment by bodedra** [comment]
cherry-pick doesn't work. Seems alias are changed in v13.0. Let me repair this PR.

File: account_invoice_margin/report/account_invoice_report.py [hunk 1]
```diff
@@ -8,7 +8,11 @@ class AccountInvoiceReport(models.Model):
...
     margin = fields.Float(string="Margin", readonly=True)
+    margin_percent = fields.Float(string="Margin (%)", readonly=True)
 
     def _select(self):
         select_str = super()._select()
-        return "%s, SUM(line.margin_signed) AS margin" % select_str
+        return (
+            "%s, SUM(line.margin_signed) AS margin, "
+            "AVG(move.margin_percent)  AS margin_percen" % select_str
+        )
```
Comment (victoralmau) [inline account_invoice_margin/report/account_invoice_report.py]:
```suggestion
        return (
            "%s, SUM(line.margin_signed) AS margin, "
            "AVG(move.margin_percent)  AS margin_percent" % select_str
        )
```

**Review by victoralmau — CHANGES_REQUESTED** [review]

File: account_invoice_margin/report/account_invoice_report.py (hunk 1 above)
Comment (bodedra) [reply account_invoice_margin/report/account_invoice_report.py]: @victoralmau comment addressed! Thanks for your review.

**Review by victoralmau — APPROVED** [review]

**Comment by bodedra** [comment]
@victoralmau still need some correction. See full details here https://github.com/OCA/margin-analysis/pull/102#pullrequestreview-729899124

File: account_invoice_margin/report/account_invoice_report.py [hunk 2]
```diff
@@ -1,14 +1,45 @@
...
+            groupby,
+            offset=offset,
+            limit=limit,
+            orderby=orderby,
+            lazy=lazy,
+        )
+        if "margin" not in fields:
+            return res
+
+        full_fields = all(x in fields for x in {"margin", "margin_percent"})
```
Comment (pedrobaeza) [inline account_invoice_margin/report/account_invoice_report.py]: Shouldn't this be?

```suggestion
        full_fields = all(x in fields for x in {"price_total", "margin_percent"})
```

File: account_invoice_margin/report/account_invoice_report.py [hunk 3]
```diff
@@ -1,14 +1,45 @@
...
+        margin_percent = "margin_percent" in fields
+        for line in res:
+            if full_fields:  # compute difference
+                line["margin_percent"] = (
+                    (line["margin"] or 0.0) / line["price_total"]
```
Comment (pedrobaeza) [inline account_invoice_margin/report/account_invoice_report.py]: Error when `line["price_total"]` = 0

Comment (bodedra) [reply account_invoice_margin/report/account_invoice_report.py]: Yes. It also problem on 12. I created invoice with 0 amount.

Comment (pedrobaeza) [reply account_invoice_margin/report/account_invoice_report.py]: Put then `if full_fields and line["price_total"]:` as condition

Comment (bodedra) [reply account_invoice_margin/report/account_invoice_report.py]: I like your solution compare to mine
> " ( line["price_total"] or 1.0) "

File: account_invoice_margin/report/account_invoice_report.py (hunk 2 above)
Comment (bodedra) [reply account_invoice_margin/report/account_invoice_report.py]: current version of fix working well as per this example: https://github.com/OCA/margin-analysis/pull/102#discussion_r688725059 or is there any additional example to verify?

I will do the same testing once runbot is up.

Comment (pedrobaeza) [reply account_invoice_margin/report/account_invoice_report.py]: The logic is that we should have here the measures that are involved in the computation, isn't it?

Comment (bodedra) [reply account_invoice_margin/report/account_invoice_report.py]: Yes, that's true. IMO both approach will give us same output. I am open to change field from `margin` to `price_total`

Comment (pedrobaeza) [reply account_invoice_margin/report/account_invoice_report.py]: Let's do it for this version then.

Comment (bodedra) [reply account_invoice_margin/report/account_invoice_report.py]: I have added as a separate commit.

**Comment by pedrobaeza** [comment]
@bodedra I'm thinking that this can be conflicting as we are accessing as `line["margin"] or 0.0` instead of `line.get("margin", 0.0)`. Why don't we put both 3 fields in the condition or switch to that way?

**Comment by bodedra** [comment]
@pedrobaeza comment addressed!

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**